README.md

```diff
@@ -8,23 +8,36 @@
 
 ## Introduction
 
-Chronos is an efficient library for asynchronous programming and an alternative to Nim's asyncdispatch.
-
-## Documentation
-You can find more documentation, notes and examples in [Wiki](https://github.com/status-im/nim-chronos/wiki).
+Chronos is an [asyncdispatch](https://nim-lang.org/docs/asyncdispatch.html)
+fork with a unified callback type, FIFO processing order for Future callbacks and [many other changes](https://github.com/status-im/nim-chronos/wiki/AsyncDispatch-comparison) that diverged from upstream's philosophy.
 
 ## Installation
-You can use Nim official package manager `nimble` to install `chronos`. The most recent version of the library can be installed via:
+You can use Nim's official package manager Nimble to install Chronos:
 
 ```
 $ nimble install https://github.com/status-im/nim-chronos.git
 ```
 
+## Documentation
+
+### Concepts
+
+Chronos implements the async/await paradigm in a self-contained library, using
+macros, with no specific helpers from the compiler.
+
+Our event loop is called a "dispatcher" and a single instance per thread is
+created, as soon as one is needed.
+
+To trigger a dispatcher's processing step, we need to call `poll()` - either
```

Comment:
Even if its possible this is not recommended way, so recommended way is to use `waitFor()` and `runForever()` only.

Reply:
It depends. If you want to exit gracefully by checking a global variable between event loop ticks, it's cleaner to use `poll()`.

Anyway, these are high-level concepts. The recommendations will come later, when we get to the API and examples.

Reply:
@stefantalpalaru with such approach you can stuck forever in `poll()` call, it is much better to use `callSoon(someCallback)` and check your variable in `someCallback`, and this will never stuck.

Reply:
If `poll()` gets stuck, callbacks are no longer being processed, right?

Reply:
For your specific use-case when you want to check some global variable for some flag and call `poll()` it can happens that if there no `callbacks` and `timers` but some descriptors are waiting (for example accept() waiting for new client) you could stuck in `poll()` until new client will not be accepted. Its why you should always use `callSoon()` and populate dispatcher's callbacks list with at least one callback so it will be called all the time... This of course very bad idea too, because in such case there will be no pauses and your CPU usage can increase significantly.

Reply:
So its why cycles like this:
```nim
while someGlobalFlagIsTrue:
  poll()
```
Is very bad idea, but my `yieldAsync` is also not a proper solution, just because it leads to infinite CPU usage.

Reply:
You're worried about this? https://en.wikipedia.org/wiki/Busy_waiting

I don't think we can avoid it, in an event loop (unless we introduce artificial pauses that affect our responsiveness), but you should know that `runForever()` does the same busy waiting: https://github.com/status-im/nim-chronos/blob/ce6e7d17b11b33d5dd17123827dda43af553a7eb/chronos/asyncloop.nim#L939-L942

Maybe we should consider `poll()` getting stuck a bug, instead of a supported state. Same for hogging the event loop with long-running tasks in the same thread.

Reply:
instead of not documenting it, it's better to document the pitfalls of using it, with a recommendation to avoid it - when it's part of public api, people will use it - less information is always worse, in such cases

Reply:
There two procedures which should be used as public API its `runForever` and `waitFor` all other procedures MUST not be used at any cost. I wish to disable `*` for `poll()` but i can't. I can write book explaining why you should not use `poll()` and most of the readers will drop this book because it will become boring. That's why documentation should avoid sentences like this:

> To trigger a dispatcher's processing step, we need to call `poll()`.

You should not call `poll()` directly at any cost, this should be present in documentation.

Reply:
How else do you stop the dispatcher for a controlled shutdown?

Reply:
Use `waitFor` if you need controlled shutdown.